Index: git/ocf/ceilometer-collector
===================================================================
--- git.orig/ocf/ceilometer-collector	2014-08-07 21:08:46.637211162 -0400
+++ git/ocf/ceilometer-collector	2014-08-07 21:09:24.893475317 -0400
@@ -223,15 +223,16 @@
         return $rc
     fi
 
-    # Check the connections according to the PID.
-    # We are sure to hit the scheduler process and not other Cinder process with the same connection behavior (for example cinder-api)
-        pid=`cat $OCF_RESKEY_pid`
-        scheduler_amqp_check=`netstat -punt | grep -s "$OCF_RESKEY_amqp_server_port" | grep -s "$pid" | grep -qs "ESTABLISHED"`
-        rc=$?
-        if [ $rc -ne 0 ]; then
+    # Check the connections according to the PID of the child process since
+    # the parent is not the one with the AMQP connection
+    ppid=`cat $OCF_RESKEY_pid`
+    pid=`pgrep -P $ppid`
+    scheduler_amqp_check=`netstat -punt | grep -s "$OCF_RESKEY_amqp_server_port" | grep -s "$pid" | grep -qs "ESTABLISHED"`
+    rc=$?
+    if [ $rc -ne 0 ]; then
         ocf_log err "Collector is not connected to the AMQP server : $rc"
         return $OCF_NOT_RUNNING
-        fi
+    fi
 
     ocf_log debug "OpenStack Ceilometer Collector (ceilometer-collector) monitor succeeded"
     return $OCF_SUCCESS
