Index: git/ocf/cinder-volume
===================================================================
--- git.orig/ocf/cinder-volume
+++ git/ocf/cinder-volume
@@ -224,6 +224,13 @@ cinder_volume_monitor() {
     pid=`cat $OCF_RESKEY_pid`
 
     if ocf_is_true "$OCF_RESKEY_multibackend"; then
+      pids=`ps -o pid --no-headers --ppid $pid`
+      rc=$?
+      if [ $rc -ne 0 ]; then
+        ocf_log err "No child processes from Cinder Volume (yet...): $rc"
+        return $OCF_NOT_RUNNING
+      fi
+
       # Grab the child's PIDs
       for i in `ps -o pid --no-headers --ppid $pid`
       do
