---
 ocf/glance-api |    3 +++
 1 file changed, 3 insertions(+)

--- a/ocf/glance-api
+++ b/ocf/glance-api
@@ -243,6 +243,9 @@ glance_api_monitor() {
         return $rc
     fi
 
+    ### DPENNEY: Bypass monitor until keyring functionality is ported
+    return $OCF_SUCCESS
+
     # Monitor the RA by retrieving the image list
     if [ -n "$OCF_RESKEY_os_username" ] && [ -n "$OCF_RESKEY_os_tenant_name" ] && [ -n "$OCF_RESKEY_os_auth_url" ]; then
 	    ocf_run -q $OCF_RESKEY_client_binary \
