Index: git/ocf/glance-api
===================================================================
--- git.orig/ocf/glance-api
+++ git/ocf/glance-api
@@ -249,7 +249,7 @@ glance_api_monitor() {
         --os_username "$OCF_RESKEY_os_username" \
         --os_tenant_name "$OCF_RESKEY_os_tenant_name" \
         --os_auth_url "$OCF_RESKEY_os_auth_url" \
-        index > /dev/null 2>&1
+        image-list > /dev/null 2>&1
         rc=$?
         if [ $rc -ne 0 ]; then
             ocf_log err "Failed to connect to the OpenStack ImageService (glance-api): $rc"
