---
 ocf/neutron-server |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/ocf/neutron-server
+++ b/ocf/neutron-server
@@ -288,7 +288,7 @@ neutron_server_start() {
     # Don't use ocf_run as we're sending the tool's output
     # straight to /dev/null anyway and using ocf_run would break stdout-redirection here.
     su ${OCF_RESKEY_user} -s /bin/sh -c "${OCF_RESKEY_binary} --config-file=$OCF_RESKEY_config \
-       	--config-file=$OCF_RESKEY_plugin_config --log-file=/var/log/neutron/server.log $OCF_RESKEY_additional_parameters"' >> \
+       	--config-file=$OCF_RESKEY_plugin_config $OCF_RESKEY_additional_parameters"' >> \
 	/dev/null 2>&1 & echo $!' > $OCF_RESKEY_pid
 
     # Spin waiting for the server to come up.
