Index: openstack-resource-agents-git-64e633d/ocf/neutron-server
===================================================================
--- openstack-resource-agents-git-64e633d.orig/ocf/neutron-server	2016-08-09 19:09:49.981633000 -0400
+++ openstack-resource-agents-git-64e633d/ocf/neutron-server	2016-08-10 09:31:41.221558000 -0400
@@ -25,6 +25,7 @@
 #   OCF_RESKEY_binary
 #   OCF_RESKEY_config
 #   OCF_RESKEY_plugin_config
+#   OCF_RESKEY_sriov_plugin_config
 #   OCF_RESKEY_user
 #   OCF_RESKEY_pid
 #   OCF_RESKEY_os_username
@@ -45,6 +46,7 @@
 OCF_RESKEY_binary_default="neutron-server"
 OCF_RESKEY_config_default="/etc/neutron/neutron.conf"
 OCF_RESKEY_plugin_config_default="/etc/neutron/plugins/openvswitch/ovs_neutron_plugin.ini"
+OCF_RESKEY_sriov_plugin_config_default="/etc/neutron/plugins/ml2/ml2_conf_sriov.ini"
 OCF_RESKEY_user_default="neutron"
 OCF_RESKEY_pid_default="$HA_RSCTMP/$OCF_RESOURCE_INSTANCE.pid"
 OCF_RESKEY_url_default="http://127.0.0.1:9696"
@@ -53,6 +55,7 @@
 : ${OCF_RESKEY_binary=${OCF_RESKEY_binary_default}}
 : ${OCF_RESKEY_config=${OCF_RESKEY_config_default}}
 : ${OCF_RESKEY_plugin_config=${OCF_RESKEY_plugin_config_default}}
+: ${OCF_RESKEY_sriov_plugin_config=${OCF_RESKEY_sriov_plugin_config_default}}
 : ${OCF_RESKEY_user=${OCF_RESKEY_user_default}}
 : ${OCF_RESKEY_pid=${OCF_RESKEY_pid_default}}
 : ${OCF_RESKEY_url=${OCF_RESKEY_url_default}}
@@ -115,6 +118,14 @@
 <content type="string" default="${OCF_RESKEY_plugin_config_default}" />
 </parameter>
 
+<parameter name="sriov_plugin_config" unique="0" required="0">
+<longdesc lang="en">
+Location of the OpenStack sriov plugin configuration file
+</longdesc>
+<shortdesc lang="en">OpenStack neutron sriov config file</shortdesc>
+<content type="string" default="${OCF_RESKEY_sriov_plugin_config_default}" />
+</parameter>
+
 <parameter name="user" unique="0" required="0">
 <longdesc lang="en">
 User running OpenStack Neutron Server (neutron-server)
@@ -288,7 +299,7 @@
     # Don't use ocf_run as we're sending the tool's output
     # straight to /dev/null anyway and using ocf_run would break stdout-redirection here.
     su ${OCF_RESKEY_user} -s /bin/sh -c "${OCF_RESKEY_binary} --config-file=$OCF_RESKEY_config \
-       	--config-file=$OCF_RESKEY_plugin_config $OCF_RESKEY_additional_parameters"' >> \
+       	--config-file=$OCF_RESKEY_plugin_config --config-file=$OCF_RESKEY_sriov_plugin_config $OCF_RESKEY_additional_parameters"' >> \
 	/dev/null 2>&1 & echo $!' > $OCF_RESKEY_pid
 
     # Spin waiting for the server to come up.
