diff --git a/ocf/nova-api b/ocf/nova-api
index 5764adc..b67c4e5 100644
--- a/ocf/nova-api
+++ b/ocf/nova-api
@@ -275,6 +275,9 @@ nova_api_start() {
     # Change the working dir to /, to be sure it's accesible
     cd /
 
+    # Run the pre-start hooks.  This can be used to trigger a nova database sync, for example.
+    /usr/bin/nova-controller-runhooks
+
     # run the actual nova-api daemon. Don't use ocf_run as we're sending the tool's output
     # straight to /dev/null anyway and using ocf_run would break stdout-redirection here.
     su ${OCF_RESKEY_user} -s /bin/sh -c "${OCF_RESKEY_binary} --config-file=$OCF_RESKEY_config \
diff --git a/ocf/nova-conductor b/ocf/nova-conductor
index dfcff97..aa1ee2a 100644
--- a/ocf/nova-conductor
+++ b/ocf/nova-conductor
@@ -294,6 +294,9 @@ nova_conductor_start() {
     # Change the working dir to /, to be sure it's accesible
     cd /
 
+    # Run the pre-start hooks.  This can be used to trigger a nova database sync, for example.
+    /usr/bin/nova-controller-runhooks
+
     # run the actual nova-conductor daemon. Don't use ocf_run as we're sending the tool's output
     # straight to /dev/null anyway and using ocf_run would break stdout-redirection here.
     su ${OCF_RESKEY_user} -s /bin/sh -c "${OCF_RESKEY_binary} --config-file=$OCF_RESKEY_config \
diff --git a/ocf/nova-scheduler b/ocf/nova-scheduler
index afaf8e9..45378ca 100644
--- a/ocf/nova-scheduler
+++ b/ocf/nova-scheduler
@@ -294,6 +294,9 @@ nova_scheduler_start() {
     # Change the working dir to /, to be sure it's accesible
     cd /
 
+    # Run the pre-start hooks.  This can be used to trigger a nova database sync, for example.
+    /usr/bin/nova-controller-runhooks
+
     # run the actual nova-scheduler daemon. Don't use ocf_run as we're sending the tool's output
     # straight to /dev/null anyway and using ocf_run would break stdout-redirection here.
     su ${OCF_RESKEY_user} -s /bin/sh -c "${OCF_RESKEY_binary} --config-file=$OCF_RESKEY_config \
