---
 ocf/neutron-server |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

--- a/ocf/neutron-server
+++ b/ocf/neutron-server
@@ -287,8 +287,11 @@ neutron_server_start() {
     # run the actual neutron-server daemon with correct configurations files (server + plugin)
     # Don't use ocf_run as we're sending the tool's output
     # straight to /dev/null anyway and using ocf_run would break stdout-redirection here.
-    su ${OCF_RESKEY_user} -s /bin/sh -c "${OCF_RESKEY_binary} --config-file=$OCF_RESKEY_config \
-       	--config-file=$OCF_RESKEY_plugin_config $OCF_RESKEY_additional_parameters"' >> \
+    ## DPENNEY: Removing plugin ref
+    ##su ${OCF_RESKEY_user} -s /bin/sh -c "${OCF_RESKEY_binary} --config-file=$OCF_RESKEY_config \
+    ##   	--config-file=$OCF_RESKEY_plugin_config $OCF_RESKEY_additional_parameters"' >> \
+    ##	/dev/null 2>&1 & echo $!' > $OCF_RESKEY_pid
+    su ${OCF_RESKEY_user} -s /bin/sh -c "${OCF_RESKEY_binary} --config-file=$OCF_RESKEY_config"' >> \
 	/dev/null 2>&1 & echo $!' > $OCF_RESKEY_pid
 
     # Spin waiting for the server to come up.
