From f6e57a8c93d4669698d86ac53e0dc57e794725a1 Mon Sep 17 00:00:00 2001
From: Al Bailey <al.bailey@windriver.com>
Date: Fri, 22 Jul 2022 19:18:44 +0000
Subject: [PATCH 3/3] Create /opt/branding as part of install

---
 debian/openstack-dashboard.dirs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/debian/openstack-dashboard.dirs b/debian/openstack-dashboard.dirs
index cb229b2..ef69f56 100644
--- a/debian/openstack-dashboard.dirs
+++ b/debian/openstack-dashboard.dirs
@@ -1,2 +1,3 @@
 /etc/openstack-dashboard/local_settings.d
 /usr/share/openstack-dashboard
+/opt/branding
-- 
2.30.2

