From 60fb7ad65595700e133e41099f9f031159540da4 Mon Sep 17 00:00:00 2001
From: Fabricio Henrique Ramos <fabriciohenrique.ramos@windriver.com>
Date: Mon, 4 Oct 2021 22:32:03 -0300
Subject: [PATCH 2/2] Disable tests

---
 .stestr.conf | 3 ---
 1 file changed, 3 deletions(-)
 delete mode 100644 .stestr.conf

diff --git a/.stestr.conf b/.stestr.conf
deleted file mode 100644
index b0e6e1ad..00000000
--- a/.stestr.conf
+++ /dev/null
@@ -1,3 +0,0 @@
-[DEFAULT]
-test_path=./oslo_messaging/tests
-top_path=./
-- 
2.17.1

