From dda42ea9d18cacb7059652e95fb1b689f175f6ac Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Wed, 20 Feb 2019 13:56:27 -0600
Subject: [PATCH 6/6] Set Min NGINX handles

---
 mariadb/files/nginx.tmpl | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mariadb/files/nginx.tmpl b/mariadb/files/nginx.tmpl
index b74b2b6..9775d89 100644
--- a/mariadb/files/nginx.tmpl
+++ b/mariadb/files/nginx.tmpl
@@ -23,7 +23,9 @@ daemon off;
 
 worker_processes {{ $cfg.WorkerProcesses }};
 pid /run/nginx.pid;
-{{ if ne .MaxOpenFiles 0 }}
+{{ if lt .MaxOpenFiles 2048 }}
+worker_rlimit_nofile 2048;
+{{else}}
 worker_rlimit_nofile {{ .MaxOpenFiles }};
 {{ end}}
 
-- 
1.8.3.1

