From 6fccedb6743ea240d52fbe52f3129e7dc5ee29db Mon Sep 17 00:00:00 2001
From: Zhipeng Liu <zhipengs.liu@intel.com>
Date: Thu, 27 Feb 2020 01:01:57 +0800
Subject: [PATCH] Fix ipv6 address issue causing mariadb ingress not ready

Signed-off-by: Zhipeng Liu <zhipengs.liu@intel.com>
---
 mariadb/files/nginx.tmpl | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mariadb/files/nginx.tmpl b/mariadb/files/nginx.tmpl
index 5ec3d0d..356fad5 100644
--- a/mariadb/files/nginx.tmpl
+++ b/mariadb/files/nginx.tmpl
@@ -473,11 +473,11 @@ stream {
     {{ range $j, $endpoint := $tcpServer.Endpoints }}
     {{ if eq $j 0 }}
         # NOTE(portdirect): see https://docs.nginx.com/nginx/admin-guide/load-balancer/tcp-health-check/#passive-tcp-health-checks to tune passive healthchecks
-        server                  {{ $endpoint.Address }}:{{ $endpoint.Port }};
+        server                  {{ formatIP $endpoint.Address }}:{{ $endpoint.Port }};
     {{ else if eq $j 1 }}
-        server                  {{ $endpoint.Address }}:{{ $endpoint.Port }} backup;
+        server                  {{ formatIP $endpoint.Address }}:{{ $endpoint.Port }} backup;
     {{ else }}
-        server                  {{ $endpoint.Address }}:{{ $endpoint.Port }} down;
+        server                  {{ formatIP $endpoint.Address }}:{{ $endpoint.Port }} down;
     {{ end }}
     {{ end }}
     }
-- 
2.7.4

